public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -224,6 +226,42 @@ export class CloudWatchDatasource extends DataSourceApi<CloudWatchQuery, CloudWa
     );
   }
 
+  private addDataLinksToResponse(response: DataQueryResponse, options: DataQueryRequest<CloudWatchQuery>) {
+    for (const dataFrame of response.data as DataFrame[]) {
+      const range = this.timeSrv.timeRange();
+      const start = range.from.toISOString();
+      const end = range.to.toISOString();
+
+      const curTarget = options.targets.find(target => target.refId === dataFrame.refId) as CloudWatchLogsQuery;
+      const urlProps: AwsUrl = {
+        end,
+        start,
+        timeType: 'ABSOLUTE',
+        tz: 'UTC',
+        editorString: curTarget.expression,
+        isLiveTail: false,
+        source: curTarget.logGroupNames,
+      };
+
+      const encodedUrl = encodeUrl(
+        urlProps,
+        this.replace(this.getActualRegion(curTarget.region), options.scopedVars, true, 'region')
```

Comment:
I think this may be similar error that I was fixing in #24346. If this is meant to interpolate the actual region in the query, `curTarget.region` should be probably interpolated first with scoped vars and then the result should be used in `getActualRegion` so that `default` gets transformed in specific region.